lib/ffaker/vehicle.rb

```diff
@@ -2,6 +2,70 @@
 
 module FFaker
   module Vehicle
+    module VIN
+      extend ModuleUtils
+      extend self
+
+      # https://en.wikibooks.org/wiki/Vehicle_Identification_Numbers_(VIN_codes)/World_Manufacturer_Identifier_(WMI)
+      VALID_WMI_REGIONS = [ ('A'..'C'), ('J'..'Z'), ('1'..'9') ].map(&:to_a).flatten
+
+      VALID_YEAR_CHARS = %w[
+        5 6 7 8 9 A B C D E F G H J K L M N P R S T V W X Y 1 2 3 4 5 6 7 8 9
+      ] # 2005-2039
+
+      # https://en.wikibooks.org/wiki/Vehicle_Identification_Numbers_(VIN_codes)/Check_digit
+      TRANSLITERATION_VALUES = {
+        A: 1, B: 2, C: 3, D: 4, E: 5, F: 6, G: 7, H: 8,
+        J: 1, K: 2, L: 3, M: 4, N: 5, P: 7, R: 9,
+        S: 2, T: 3, U: 4, V: 5, W: 6, X: 7, Y: 8, Z: 9
+      }
+      POSITION_WEIGHTS = [8, 7, 6, 5, 4, 3, 2, 10, 0, 9, 8, 7, 6, 5, 4, 3, 2]
+
+      VALID_ALPHA = TRANSLITERATION_VALUES.keys.map(&:to_s)
+      VALID_ALPHANUMERIC = VALID_ALPHA + ('1'..'9').to_a
+
+      # Generate a VIN that is compliant with specifications of US Title 49 Section 565.15
+      # https://www.govinfo.gov/content/pkg/CFR-2019-title49-vol6/xml/CFR-2019-title49-vol6-part565.xml#seqnum565.15
+      #
+      # Position  Meaning
+      # 1-3       Manufacturer ID aka WMI (alpha and digits)
+      # 4-8       Vehicle Description ("For passenger cars ..[position 7].. shall be alphabetic")
+      # 9         Check Digit (0-9 or "X")
+      # 10        Year (see VIN_VALID_YEARS)
+      # 11        Plant of manufacture (alpha and digits)
+      # 12-17     Serial number (digits only)
+      #
+      # I, O and Q are NOT allowed. VIN_VALID_ALPHA has valid alpha characters.
+      def vin
+        generated_vin = [
+          [ # Manufacturer ID / WMI
+            fetch_sample(VALID_WMI_REGIONS),
+            2.times.map{ fetch_sample(VALID_ALPHANUMERIC) }.join
+          ].join,
+          [ # Vehicle Description
+            3.times.map{ fetch_sample(VALID_ALPHANUMERIC) }.join,
+            fetch_sample(VALID_ALPHA),
+            fetch_sample(VALID_ALPHANUMERIC)
+          ].join,
+          '0', # check digit placeholder
+          fetch_sample(VALID_YEAR_CHARS), # Year of Manufacture
+          fetch_sample(VALID_ALPHANUMERIC), # Plant ID
+          6.times.map{ rand(10).to_s }.join # Serial number
+        ].join
+
+        # Calculate the Check Digit
+        weighted_sum = generated_vin.chars.each_with_index.map{|char, idx|
+          (TRANSLITERATION_VALUES[char.to_sym] || char).to_i * POSITION_WEIGHTS[idx]
+        }.reduce(:+)
```

Comment:
I believe map ... reduce(:+) can replace with sum ...

```rb
[2] pry(main)> (1..10).map { |i| i*2 }.reduce(:+)
=> 110
[3] pry(main)> (1..10).sum { |i| i*2 }
=> 110
```

```suggestion
        weighted_sum = generated_vin.chars.each_with_index.sum { |char, idx|
          (TRANSLITERATION_VALUES[char.to_sym] || char).to_i * POSITION_WEIGHTS[idx]
        }
```